### ISSUE TITLE
Option to have close buttons present on each of the tabs yet only visible on the hovered tab

### ISSUE THREAD

**Issue by jdegeete** [mborgerson/pyside6_qtads#32]
I am wondering whether there is an easy solution to provide this functionality. I have currently achieved this in my pyside6 application in the most horrible way by:
1) Enabling the close button for each tab.
2) Having two icon images for the close QPushButton. One having the same color as the background.
3) By Installing a (global) event filter on the app. This event filter catches 'enter' and 'leave' events on the tab and changes the style of the QPushButton accordingly. Without this, the color of the QPushButton only changes by going over the QPushButton rather than the tab.

**Comment by jeandet** [comment]
I think this is possible providing your own CSS to override [this](https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/blob/master/src/stylesheets/default.css#L86-L103)

**Comment by jdegeete** [comment]
Yes, I used that. However, the problem is that the cursor needs to be directly above the close button. Hovering over the tab does not trigger the hover over the close button.

I solved this by doing the following:

```
class GlobalEventFilter(QObject):
    def eventFilter(self, obj, event):
        if isinstance(obj, QtAds.CDockWidgetTab):
            if event.type() == QtCore.QEvent.Enter:
                obj.findChild(QPushButton, "tabCloseButton").setStyleSheet("QPushButton { icon: url(./resources/close-button2.svg); }")
            elif event.type() == QtCore.QEvent.Leave:
                obj.findChild(QPushButton, "tabCloseButton").setStyleSheet("QPushButton { icon: url(./resources/close-button.svg); }")
        return super().eventFilter(obj, event)
        
        
if __name__ == '__main__':
    app = QApplication(sys.argv)
    app.setStyleSheet(stylesheet)
    app.installEventFilter(GlobalEventFilter(app))
    window.show()
    sys.exit(app.exec())
```

This is a very crude solution.

**Comment by jeandet** [comment]
@jdegeete I see, I didn't try it but maybe you can get something with `ads--CDockWidgetTab:hover QPushButton`, (I'm not a qss expert).

**Comment by jdegeete** [comment]
@jeandet I had already tried that with no effect so I am assuming that qss does not allow it (also saw nothing in the documentation that hinted at it being possible).
